bee-message/src/message.rs

```diff
@@ -0,0 +1,195 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{parents::Parents, payload::Payload, MessageId, MessagePackError, MessageUnpackError, ValidationError};
+
+use bee_packable::{PackError, Packable, Packer, UnpackError, Unpacker};
+
+use crypto::{
+    hashes::{blake2b::Blake2b256, Digest},
+    signatures::ed25519,
+};
+
+use alloc::vec::Vec;
+use core::{convert::TryInto, ops::RangeInclusive};
+
+/// Range (in bytes) of a valid message length.
+pub const MESSAGE_LENGTH_RANGE: RangeInclusive<usize> = 53..=32768;
+
+/// Length (in bytes) of a public key.
+pub const MESSAGE_PUBLIC_KEY_LENGTH: usize = 32;
+
+/// Length (in bytes) of a message signature.
+pub const MESSAGE_SIGNATURE_LENGTH: usize = 64;
+
+/// Messages are of version 1.
+const MESSAGE_VERSION: u8 = 1;
+
+/// Represents the object that nodes gossip around the network.
+///
+/// `Message`s must:
+/// * Have a length (in bytes) within `MESSAGE_LENGTH_RANGE`.
+/// * Ensure all applicable data is appropriately validated.
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "enable-serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Message {
+    /// Message [Parents].
+    pub(crate) parents: Parents,
+    /// The public key of the issuing node.
+    pub(crate) issuer_public_key: [u8; MESSAGE_PUBLIC_KEY_LENGTH],
+    /// The Unix timestamp at the moment of issue.
+    pub(crate) issue_timestamp: u64,
+    /// The sequence number of the message, indicating the marker sequence it belongs to.
+    pub(crate) sequence_number: u32,
+    /// The optional [Payload] of the message.
+    pub(crate) payload: Option<Payload>,
+    /// The result of the Proof of Work in order for the message to be accepted into the tangle.
+    pub(crate) nonce: u64,
+    /// Signature signing the above message fields.
+    #[cfg_attr(feature = "enable-serde", serde(with = "serde_big_array::BigArray"))]
+    pub(crate) signature: [u8; MESSAGE_SIGNATURE_LENGTH],
+}
+
+impl Message {
+    /// Computes the identifier of the message.
+    pub fn id(&self) -> (MessageId, Vec<u8>) {
```

Comment:
I would just return the `MessageId`. This was a weird API that I'd like to avoid in the future.

Reply:
Sure 👍